### PR TITLE
Fix SPI preproc guards

### PR DESCRIPTION
**Pull request by caternuson**
For #66 

Tested on a Adafruit ProxLight Trinkey with:
```cpp
#include "Adafruit_APDS9960.h"

//the pin that the interrupt is attached to
#define INT_PIN PIN_INTERRUPT

//create the APDS9960 object
Adafruit_APDS9960 apds;

void setup() {
  Serial.begin(115200);
  while(!Serial);
  pinMode(INT_PIN, INPUT_PULLUP);

  if(!apds.begin()){
    Serial.println("failed to initialize device! Please check your wiring.");
  }
  else Serial.println("Device initialized!");

  //enable proximity mode
  apds.enableProximity(true);

  //set the interrupt threshold to fire when proximity reading goes above 175
  apds.setProximityInterruptThreshold(0, 175);

  //enable the proximity interrupt
  apds.enableProximityInterrupt();
}

void loop() {

  //print the proximity reading when the interrupt pin goes low
  if(!digitalRead(INT_PIN)){
    Serial.println(apds.readProximity());

    //clear the interrupt
    apds.clearInterrupt();
  }
}
```

Screenshot from 2021-09-28 12-17-54

### CODE REVIEW

**Review by ladyada — APPROVED** [review]

**Review by ladyada — APPROVED** [review]

**Comment by caternuson** [comment]
The initial commit in the PR is a red herring. Can fix #66 with just the addition of `#include <Arduino.h>` to `Adafruit_SPIDevice.h`. Without that, the logic to check for SPI was not working and the contents of the file retained, so failed at compile time.

Added all the new logic to `Adaffruit_BusIO_Register` to get those examples to compile locally for Adafruit ProxLight Trinkey. That's not part of the CI checks, so might be overkill?

**Comment by ladyada** [comment]
up to you

**Comment by caternuson** [comment]
My pref is to remove for now. This was a quick hack. Can add later when needed.

**Comment by caternuson** [comment]
OK. CI's happy.

Prox Trinkey is happy
Screenshot from 2021-09-28 15-21-01

I'm happy removing the hacks-of-desperation from `Adaffruit_BusIO_Register`.

TLDR issue was a subtle Arduino "feature" that took waaaaaaaaaay too long to spot.

merging
.

**Comment by ladyada** [comment]
it happens! sometimes you go all the way around to end up where ya started, thanks for being diligent
